components/common/site_name_and_description.jsx

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+export default class SiteNameAndDescription extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * Custom description of the site
+         */
+        customDescriptionText: PropTypes.string,
+
+        /**
+         * Set to true if instance is licensed; false as default
+         */
+        isLicensed: PropTypes.bool,
+
+        /**
+         * Site name; "Mattermost" as default
+         */
+        siteName: PropTypes.string,
+    };
+
+    static defaultProps = {
+        isLicensed: false,
+        siteName: 'Mattermost',
+    };
+
+    render() {
+        const {
```

Comment:
Would this component be more simply expressed as a pure function, with this destructuring occurring more naturally as:
```
const SiteNameAndDescription = ({customDescriptionText, isLicensed, siteName}) => {
    // ...
};
```

Reply:
I like that approach as well but `eslint` will prompt us with `missing props validation`, so it will end up like:
```
const SiteNameAndDescription = ({customDescriptionText, isLicensed, siteName}) => {
    ...
};

SiteNameAndDescription.propTypes = {
    ...
};

SiteNameAndDescription.defaultProps = {
   ...
};
```

That's why I'm sticking into this pattern.

Reply:
IMO, I think the style you posted above is still a net win, and solves the props definition cleanly. 0/5 on changing this, but I'm curious if we have a coding style that prohibits the above?

Reply:
Cool, I'll take note of this.
I believe it's `"react/prop-types": [2, { "ignore": ["match", "location", "history", "component"] }],`